pkg/splunk/client/minioclient.go

```diff
@@ -47,14 +47,14 @@ type MinioClient struct {
 }
 
 // NewMinioClient returns an Minio client
-func NewMinioClient(ctx context.Context, bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, region string, endpoint string, fn GetInitFunc) (S3Client, error) {
+func NewMinioClient(ctx context.Context, bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, region string, endpoint string, fn GetInitFunc) (RemoteDataClient, error) {
```

Comment:
Same here, instead of passing so many arguments, should we consider using structure of `MinioBucketCredentials`

Reply:
Like the idea. Let me revisit this in one of the upcoming sprints if that's ok with you.